pyo3-derive-backend/src/pyclass.rs

```diff
@@ -208,11 +208,9 @@ fn parse_descriptors(item: &mut syn::Field) -> syn::Result<Vec<FnType>> {
     Ok(descs)
 }
 
-/// The orphan rule disallows using a generic inventory struct, so we create the whole boilerplate
-/// once per class
-fn impl_inventory(cls: &syn::Ident) -> TokenStream {
-    // Try to build a unique type that gives a hint about it's function when
-    // it comes up in error messages
+/// To allow multiple #[pymethods] block, we define inventory types.
+fn impl_methods_inventory(cls: &syn::Ident) -> TokenStream {
+    // Try to build a unique type for better error messages
     let name = format!("Pyo3MethodsInventoryFor{}", cls);
```

Comment:
Looking at the comment "The orphan rule disallows using a generic inventory struct..."

It might be possible now to achieve this, as `Rust 1.41.0` relaxed orphan rules slightly: https://blog.rust-lang.org/2020/01/30/Rust-1.41.0.html#relaxed-restrictions-when-implementing-traits

So maybe we can have `Pyo3MethodsInventory<MyClass>`

Reply:
What the above change made possible is `impl<T> ForeignTrait<LocalType> for ForeignType<T>`.
`impl inventory::Collect for PyO3MethodsInventory<MyClass>` is still impossible.

Reply:
Ah yep you're correct 👍